circuits/cpp/src/aztec3/circuits/hash.hpp

```diff
@@ -55,7 +55,7 @@ template <typename NCT> typename NCT::fr compute_partial_contract_address(typena
     std::vector<fr> const inputs = {
         fr(0), fr(0), fr(0), fr(0), contract_address_salt, function_tree_root, constructor_hash,
     };
-    return NCT::compress(inputs, aztec3::GeneratorIndex::PARTIAL_CONTRACT_ADDRESS);
+    return NCT::hash(inputs, aztec3::GeneratorIndex::PARTIAL_CONTRACT_ADDRESS);
```

Comment:
@suyash67 I understand that changing from `compress` (homomorphic) to `hash` would be a problem if were computing the full contract address as: 

```
address1 = pedersen(pubkey, contract_address_salt, function_tree_root, constructor_hash)
```

Because then address1 would be different from address2 below if `pedersen` were non-homomorphic, right?
```
partial_address = pedersen(0, contract_address_salt, function_tree_root, constructor_hash)
address2 = pedersen(pubkey, partial)
```

However, since we are always calculating the full address just by hashing twice, it doesn't matter whether the flavor is homomorphic or not, correct?